### ISSUE TITLE
rule: nonreplaced-no-aspect-ratio

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#131]
## Rule ID
`nonreplaced-no-aspect-ratio`

## What does this rule detect?
On non-replaced inline elements (`display: inline`), `aspect-ratio` has no effect because inline elements don't have controllable dimensions.

## Example: should warn
```html
<span style="display: inline; aspect-ratio: 16/9;">text</span>
```

## Example: should NOT warn
```html
<img style="aspect-ratio: 16/9;" src="photo.jpg" />
<div style="aspect-ratio: 16/9;">content</div>
```

## Related CSS Properties
`display`, `aspect-ratio`

## References
- [CSS Box Sizing Level 4 §5.1 - aspect-ratio](https://drafts.csswg.org/css-sizing-4/#aspect-ratio): requires at least one auto dimension to take effect
- Increasingly relevant as `aspect-ratio` adoption grows
